lib/src/room.dart

```diff
@@ -1187,22 +1187,26 @@ class Room {
   /// Set the power level of the user with the [userID] to the value [power].
   /// Returns the event ID of the new state event. If there is no known
   /// power level event, there might something broken and this returns null.
-  Future<String> setPower(String userID, int power) async {
-    final powerMap = Map<String, Object?>.from(
-      getState(EventTypes.RoomPowerLevels)?.content ?? {},
-    );
+  Future<String> setPower(String userId, int power) async {
+    final powerLevelMapCopy =
+        getState(EventTypes.RoomPowerLevels)?.content.copy() ?? {};
+
+    final usersPowerMap =
+        powerLevelMapCopy.tryGetMap<String, int>('users') ?? {};
 
-    final usersPowerMap = powerMap['users'] is Map<String, Object?>
-        ? powerMap['users'] as Map<String, Object?>
-        : (powerMap['users'] = <String, Object?>{});
+    if (power == getDefaultPowerLevel(powerLevelMapCopy)) {
```

Comment:
Couldn't find anything in the spec which suggests we should do this, but probably fine. Can we add a test for this as well? :D